macaddress.go

```diff
@@ -0,0 +1,47 @@
+// vim:ts=2:sw=2:et:ai:sts=2
+package main
+
+import (
+	"errors"
+	"net"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	bcastMAC = net.HardwareAddr{0xff, 0xff, 0xff, 0xff, 0xff, 0xff}
+	localMAC = net.HardwareAddr{0x00, 0xb0, 0x52, 0x00, 0x00, 0x01}
+)
+
+type macAddressValue net.HardwareAddr
+
+func (m *macAddressValue) Set(s string) error {
+	if s == "broadcast" || s == "all" {
+		*m = (macAddressValue)(bcastMAC)
+	} else if s == "local" {
+		*m = (macAddressValue)(localMAC)
+	} else {
+		v, err := net.ParseMAC(s)
+		if err != nil {
+			return err
+		} else if len(v) != 6 {
+			return errors.New("Invalid address length")
```

Comment:
Nit: most Go style guides recommend starting error strings with lower case, since they are often cited within log messages.
```suggestion
			return errors.New("invalid address length")
```

Reply:
good to know!